[16.0][MIG] stock_barcodes: Migration to 16.0

chienandalu opened the pull request:
Followup to #587

- Drop scan read logs feature, which use was deprecated already in the
  previous versions.
- Pass bootstrap migration scripts.
- Fix views issues.
- Other code fixes.

cc @Tecnativa

@FranzPoize I've been doing the migration with @sergio-teruel, who's the main developer of the module in the spanish OCA days. We started from your migration head.

At this point the module is completely functional, although we're still tweaking the views to fit the previous state.

sergio-teruel committed:
[ADD] stock_barcodes: New module to read barcodes on stocks operations (#195)

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update stock_barcodes.pot

astirpe committed:
[12.0][MIG] stock_barcodes

sergio-teruel committed:
[ADD] stock_barcodes: Allow to read barcodes from pickings

pedrobaeza committed:
[FIX] stock_barcodes: Fix ES translation

sergio-teruel committed:
[ADD] stock_barcodes: Allow to read barcodes from pickings. Last changes

sergio-teruel committed:
[FIX] stock_barcodes: Fix validate picking from candidate picking kanban view

sergio-teruel committed:
[IMP] stock_barcodes: Hidden lot for non tracking products.
When user creates a lot, set default product the wizard selected product.
When confirm the lot creation wizard, set this lot in lot wizard scanning barcode field.

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-11.0/stock-logistics-barcode-11.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-11-0/stock-logistics-barcode-11-0-stock_barcodes/

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-stock_barcodes/

chienandalu committed:
[FIX] stock_barcodes: out picking location workflow

When we've got an out picking, the logical workflow is that the scanned location
is the location we're getting the stock from

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 12.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-12.0/stock-logistics-barcode-12.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-12-0/stock-logistics-barcode-12-0-stock_barcodes/

chienandalu committed:
[FIX] stock_barcodes: location_dest_id

OCA-git-bot committed:
stock_barcodes 12.0.1.1.1

sergio-teruel committed:
[IMP] stock_barcodes: black, isort

sergio-teruel committed:
[MIG] stock_barcodes: Migration to v13.0

oca-travis committed:
[UPD] Update stock_barcodes.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

pedrobaeza committed:
[FIX] stock_barcodes: Make tests more resilient

UID = 1 is sometimes called "System" and sometimes "OdooBot", depending on the
installed set of modules, so tests can fail if you rely on this data.

Removing that variable from the tests, we get resilient tests.

pedrobaeza committed:
[FIX] stock_barcodes: Another non resilient test

Followup of dab52c9. There's another test that hardcodes OdooBot name for uid=1.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

sergio-teruel committed:
[IMP] stock_barcodes: Add barcode field to location view in same group like as v12.0

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

chienandalu committed:
[IMP] stock_barcodes: allow to add lines even if not qtys reserved

If we had any reservation (i.e unreserve), the barcodes button was
hidden.

sergio-teruel committed:
[ADD] stock_barcodes: Add option to get lots automatically in inventory depends on removal strategy
TT27844

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_barcodes 13.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-13.0/stock-logistics-barcode-13.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-13-0/stock-logistics-barcode-13-0-stock_barcodes/

fshah-initos committed:
[IMP] stock_barcodes: black, isort, prettier

fshah-initos committed:
[14.0][MIG]stock_barcodes

etobella committed:
[FIX] stock_barcodes: use the uom of the package if defined

etobella committed:
[IMP] stock_barcodes: Apply putaway strategy

chienandalu committed:
[FIX] stock_barcodes: avoid mixing tracked and untracked quants

Using the auto lot feature in the inventory scanning we want to avoid
distributing quantities to quants with no lot if the products if the
product change from untracked to tracked.

TT31267

chienandalu committed:
[FIX] stock_barcodes: reset lot id if we change the product manually

Otherwise the previous product lot would persist

TT31267

oca-travis committed:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] stock_barcodes: Refactoring code and improvements

OCA-git-bot committed:
stock_barcodes 14.0.1.1.0

mariadforgeflow committed:
[FIX] stock_barcodes: make available the action to read barcodes

OCA-git-bot committed:
stock_barcodes 14.0.1.2.0

LoisRForgeFlow committed:
[FIX] stock_barcodes: take company from env not from user

OCA-git-bot committed:
stock_barcodes 14.0.1.2.1

LoisRForgeFlow committed:
[FIX] stock_barcodes: read act windows permission issue

OCA-git-bot committed:
stock_barcodes 14.0.1.2.2

LoisRForgeFlow committed:
[FIX] stock_barcodes: set option to create lot to true in test

It was de default behavior in GS1 before the refactor

LoisRForgeFlow committed:
[FIX] stock_barcodes: several small adjustments.

* When a picking is processed partially it was not
  posible to validate it because the backorder wizard
  was captured and never returned to the user.
* Only update sml locations when there is a location
  to update, otherwise constraints for non null values
  could raise.
* Invalidate cache of todo lines when determining
  next action to do to ensure that line status is
  updated.
* Reword string in button from "Force" to "Done"
  as it was a bit misleading. Actually, in spanish
  translation it was already being translated to
  "Hecho" ("Done").

LoisRForgeFlow committed:
[IMP] stock_barcodes: show quantity even when not editable.

LoisRForgeFlow committed:
[MIG] stock_barcodes: clean warnings and adapt to v14.

Adapt last refector done in v13 to v14.

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 14.0.2.0.0

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/

francesco-ooops committed:
Added translation using Weblate (Italian)

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 35.6% (94 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 35.9% (95 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (264 of 264 strings)

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/it/

LoisRForgeFlow committed:
[FIX] stock_barcodes: make actions readable by non-admin users.

leftover from migration.

LoisRForgeFlow committed:
[FIX] stock_barcodes: do not use PO UoM by default

When creating new stock move lines, we should use the UoM in the
move, if any, or the default UoM of the product.

OCA-git-bot committed:
stock_barcodes 14.0.2.0.1

LoisRForgeFlow committed:
[FIX] stock_barcodes: remove last scan do not revert last action.

In pickings, the scan log was always generated without details of
the move lines modified/created. Without this information in the log
removing the log was dong nothing else than just that, removing the
log; leaving the last operation still reflected in the picking.

Also, when the picking was a draft one, the stock move line and
the stock move could have been created by the scan itself, so we
need to remove it in that case, reverting the action would leave
the stock move there that could be validated by mistake.

LoisRForgeFlow committed:
[FIX] stock_barcodes: use float_compare to avoid rounding issues

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 14.0.2.0.2

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-14.0/stock-logistics-barcode-14.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-14-0/stock-logistics-barcode-14-0-stock_barcodes/

sergio-teruel committed:
[IMP] stock_barcodes: Clean wizard values after the move to do has been reset

OCA-git-bot committed:
stock_barcodes 14.0.2.0.3

etobella committed:
[IMP] stock_barcodes: black, isort, prettier

etobella committed:
[MIG] stock_barcodes: Migration to 15.0

mariadforgeflow committed:
[IMP] stock_barcodes: add Barcode Option Group for not planned pickings

LoisRForgeFlow committed:
[MIG/FIX] stock_barcodes: fix notification usage to adapt to v15.

LoisRForgeFlow committed:
[IMP] stock_barcodes: attend new pylint mandatory checks

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
stock_barcodes 15.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

MiquelRForgeFlow committed:
[FIX] stock_barcodes: add migration script

sergio-teruel committed:
[FIX] stock_barcodes: Do not create a lot with create lot option enabled for products without tracking

sergio-teruel committed:
[FIX] stock_barcodes: Pending moves are not displayed when moves origin is stock_moves without reservations

OCA-git-bot committed:
stock_barcodes 15.0.1.2.0

sergio-teruel committed:
[FIX] stock_barcodes: put in pack call method fails

OCA-git-bot committed:
stock_barcodes 15.0.1.2.1

sergio-teruel committed:
[IMP] stock_barcodes: take into account package fields when scan barcodes

sergio-teruel committed:
[FIX] stock_barcodes: Not pending moves when origin pending moves is stock move lines
TT41789

OCA-git-bot committed:
stock_barcodes 15.0.1.2.2

sergio-teruel committed:
[IMP] stock_barcodes: Add prepare method to set wizard values from picking to be more inheritable

pere-aquarian committed:
Translated using Weblate (Spanish)

Currently translated at 69.1% (175 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Allow display detailed operations in barcode wizard

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Display partner on wizard

sergio-teruel committed:
[IMP] stock_barcodes: Do not clean values forced in an action done

sergio-teruel committed:
[IMP] stock_barcodes: add package to moves todo group key

sergio-teruel committed:
[IMP] stock_barcodes: display picking notes in wizard barcode

sergio-teruel committed:
[IMP] stock_barcodes: display improvements

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

carlosdauden committed:
[IMP] stock_barcodes: Show same type ready pickings tree after validate picking

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Allow make inventories

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[FIX] stock_barcodes: Check serial product tracking for inventory when a quant is created

sergio-teruel committed:
[IMP] stock_barcodes: Improve heritability of write on stock move lines

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] stock_barcodes: Better inheritance of qty focus

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Add hook to process values when moves todo have been created.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.0% (248 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Test improvements and create lots in confirm action

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (253 of 253 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

Commit:
[UPD] Update stock_barcodes.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (276 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/es/

sergio-teruel committed:
[IMP] stock_barcodes: Filled all fields if they is defined in options as filled default.
Determine scan step after determine todo action

sergio-teruel committed:
[FIX] stock_barcodes: Incorrect order of detailed operations fixed

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] Add security group to barcodes main menu

sergio-teruel committed:
[IMP] stock_barcodes: Allow skip clean values when a sml is created or updated in action_done method.

sergio-teruel committed:
[IMP] stock_barcodes: Filter sml lines bby complete packages

sergio-teruel committed:
[IMP] stock_barcodes: Display only the quants pending to read. Allow set to zero

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Add option to keep wizard values until the pending move is completed

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Try fill wizard data after product is read if fill_fields_from_lot is checked

sergio-teruel committed:
[IMP] stock_barcodes: Add total product qty without lot, package, etc

sergio-teruel committed:
[IMP] stock_barcodes: Set order in inventory quants

CarlosRoca13 committed:
[IMP] stock_barcodes: Visual improvements on footer

sergio-teruel committed:
[FIX] stock_barcodes: Reload todo action after unlink sml done

sergio-teruel committed:
[IMP] stock_barcodes Set to readonly total qty fields

sergio-teruel committed:
[IMP] stock_barcodes: Recompute fill_todo_records after unlink sml.

sergio-teruel committed:
[IMP] stock_barcodes: make todo qty_done compute readonly to better computation of pending lines

CarlosRoca13 committed:
[IMP] stock_barcodes: Responsive dropdown

sergio-teruel committed:
[IMP] stock_barcodes: Return stock move line created or updated after confirm action

carlosdauden committed:
[IMP] stock_barcodes: Ignore rest instead of Done to avoid that user skip pending operations

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

sergio-teruel committed:
[IMP] stock_barcodes: Inject a context to know when a picking is validated from barcodes interface

sergio-teruel committed:
[FIX] stock_barcodes: Set incorrect package and owner in set info from quants

sergio-teruel committed:
[FIX] stock_barcodes: create new lines when qty_done is greather than initial demand
TT46503

sergio-teruel committed:
[FIX] stock_barcodes: qty_done is increased if user scan a package twice

sergio-teruel committed:
[FIX] stock_barcodes: Link to correct move whe a sml is created

sergio-teruel committed:
[FIX] stock_barcodes: Apply locations domain in pickings and inventory

sergio-teruel committed:
[FIX] stock_barcodes: Always re-write las location scanned is not candidate sml's was found

sergio-teruel committed:
[IMP] stock_barcodes: Add context to avoid update quantities from lot if fill data from lot is active

sergio-teruel committed:
[IMP] stock_barcodes: Search packages in all internal locations is forced option is not checked

sergio-teruel committed:
[FIX] stock_barcodes: The package is not taken into account when user does an inventory

carlosdauden committed:
[IMP] stock_barcodes: Add accumulate_read_quantity option to be used in inventory. In the future it can be used by other process.
TT46934

carlosdauden committed:
[IMP] stock_barcodes: Allow find package in external locations. Add barcode_domain_field context
TT46823

carlosdauden committed:
[IMP] stock_barcodes: Fill inventory quantity when scan package
TT46934

mymage committed:
Translated using Weblate (Italian)

Currently translated at 90.2% (249 of 276 strings)

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/it/

carlosdauden committed:
[IMP] stock_barcodes: Add option ignore_quant_location
TT47469

sergio-teruel committed:
[FIX] stock_barcodes: product not take into account to search quants.

sergio-teruel committed:
[IMP] stock_barcodes: Limit product quantities entry to avoid barcode read errors over an input element
TT48037

FranzPoize committed:
[MIG] stock_barcodes: Migration to 16.0

sergio-teruel committed:
[IMP] stock_barcodes: Allow prefill product_qty when not in manual entry mode
TT48077

sergio-teruel committed:
[IMP] stock_barcodes: Add context to skip set info from quants although the configuration option is activated

sergio-teruel committed:
[IMP] stock_barcodes: Allow display notifications to web client

sergio-teruel committed:
[IMP] stock_barcodes: Allow apply putaway strategy as screen option

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

sergio-teruel committed:
[FIX] stock_barcodes: Required lot_id although lot_name is set without create lot option is checked.
Display notification for required field.

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

sergio-teruel committed:
[IMP] stock_barcodes: Skip backorder creation for pickings with all moves read or ignore pending quantities

sergio-teruel committed:
[IMP] stock_barcodes: Add option group to allow to select if you want order by location or location_dest_id

Commit:
[UPD] Update stock_barcodes.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-barcode-15.0/stock-logistics-barcode-15.0-stock_barcodes
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-barcode-15-0/stock-logistics-barcode-15-0-stock_barcodes/

FranzPoize committed:
[IMP] stock_barcodes: black, isort, prettier

chienandalu committed:
[MIG] stock_barcodes: followup migration

TT44219

FranzPoize committed:
[MIG] stock_barcodes: more migrating

sergio-teruel committed:
[FIX] stock_barcodes: Do not search quant when a only lot is scanned
TT49023

chienandalu committed:
[MIG] stock_barcodes: continue migration

- Drop scan read logs feature, which use was deprecated already in the
  previous versions.
- Pass bootstrap migration scripts.
- Fix views issues.
- Other code fixes.

pedrobaeza commented:
This doesn't contain #604

pedrobaeza commented:
/ocabot migration stock_barcodes

chienandalu commented:
> This doesn't contain #604

Fixed

FranzPoize commented:
Hello ! I pushed some stuff just after you created this PR I think. In it there was a lot of work to replace the `always_reload` feature of old odoo version.

chienandalu commented:
Hi, @FranzPoize !

Do you mean this new stuff or something else?: https://github.com/OCA/stock-logistics-barcode/pull/587/commits/38288280b173ffc2487758a85edad3bb74959442#diff-f69121127ec9ebdc28296fa3a7f3760d71865de0267308101bf6d467fef4769cR106-R121

FranzPoize commented:
No I mean this stuff https://github.com/OCA/stock-logistics-barcode/commit/38288280b173ffc2487758a85edad3bb74959442#diff-00871f58b13dd92c5ff7599bc210323e9a6e64bdbf58abd5a3c19c83428d5dbb

What you linked to is some other features we added:
* Allowing scanning product on at a time in new picking (not use the total quant by default)
* Forbidding picking validation if their is a move_line with the same location_id and location_dest_id
* Offer an option to not check demand during picking (this solve the issue where if you reload after a picking creation you will not be able to pick more product without using the force action done)
* Update all destination of all move_line when scanning a location (This allows destination to be scanned last because otherwise it does not work)

chienandalu commented:
> there was a lot of work to replace the always_reload feature of old odoo version.
> I mean this stuff https://github.com/OCA/stock-logistics-barcode/commit/38288280b173ffc2487758a85edad3bb74959442#diff-00871f58b13dd92c5ff7599bc210323e9a6e64bdbf58abd5a3c19c83428d5dbb

Can you elaborate on why this is needed? What's the issue you're trying to solve?

> other features we added:
> - Allowing scanning product on at a time in new picking (not use the total quant by default)
> - Forbidding picking validation if their is a move_line with the same location_id and location_dest_id
> - Offer an option to not check demand during picking (this solve the issue where if you reload after a picking creation you will not be able to pick more product without using the force action done)
> - Update all destination of all move_line when scanning a location (This allows destination to be scanned last because otherwise it does not work)

I commented these with @sergio-teruel last week and some of them could be accomplished with the current config (I'm sure he can elaborate in detail :)). In any case, is better to limit the scope the PR to the module migration stuff and do any feature proposal in a later one

FranzPoize commented:
You don't have to take those features if you don't want to. But you should take the `always_reload` feature imho because otherwise there is no update of many2one field unless there is a button pressed.

FranzPoize commented:
In 14.0 there was this https://github.com/OCA/stock-logistics-barcode/blob/fe5c849fc9a6ec190eed2672086df22553c78d64/stock_barcodes/wizard/stock_barcodes_read_picking_views.xml#L160 This does not work anymore the code here https://github.com/OCA/stock-logistics-barcode/commit/38288280b173ffc2487758a85edad3bb74959442#diff-00871f58b13dd92c5ff7599bc210323e9a6e64bdbf58abd5a3c19c83428d5dbb replaces this always_reload feature.

And I said many2ones in my previous comment I made a mistake. This allows one2many fields and many2many field to update their data when an onchange event is triggered.

chienandalu commented:
Hi @FranzPoize let me check it and I tell you

sergio-teruel committed:
[IMP] stock_barcodes: Refresh data. Change dependencies and call bus to reload wizard data

sergio-teruel committed:
[FIX] stock_barcodes: todo lines not refresh data for automatic mode when a new stock move line is created

lk-eska commented:
Is there any progress on this PR?

sergio-teruel commented:
Hi @ik-eska two commits 4 days ago... Any progress?? Yes...

lk-eska commented:
Sorry @sergio-teruel I haven't noticed them.

sergio-teruel commented:
jaja... don't worry... :smile:

sergio-teruel committed:
[FIX] stock_barcodes: The field state from stock move line compared is not correct
TT49399

sergio-teruel commented:
Cherry pick from 15.0 ef65da1c6e432e37dd045a77314e79c74a60fe9b

sergio-teruel committed:
[FIX] stock_barcodes: Notify events not managed by resource

sergio-teruel committed:
[FIX] stock_barcodes: Crash if user defines an option to scan with a field which not exists

chienandalu committed:
[FIX] stock_barcodes: styling

pedrobaeza commented:
@chienandalu @sergio-teruel please fix the CI and finish this for merging it.

josepmy commented:
Se puede ayudar en algo para avanzar el merge?

alexis-via committed:
[IMP] stock_barcodes: inherit create in batch to avoid warning

carlosdauden committed:
[IMP] stock_barcodes: Add pending move with difference between demand and reserved

carlosdauden committed:
[IMP] stock_barcodes: Performance and extra checks from 16.0

carlosdauden committed:
[IMP] stock_barcodes: Avoid NewId environment calling method on_barcode_scanned instead of onchange
Onchange limits documented in method:
It is not possible for a ``one2many`` or ``many2many`` field to modify
itself via onchange. This is a webclient limitation - see `#2693 <https://github.com/odoo/odoo/issues/2693>`_.

This causes refresh problems when creating or modifying records, and it is also not ideal to create records in "onchange mode"

pedrobaeza commented:
@sergio-teruel please put this green to be merged.

carlosdauden committed:
[FIX] stock_barcodes: Remove dead code

sergio-teruel committed:
[FIX] stock_barcodes: Fix tests

sergio-teruel commented:
@pedrobaeza Tests ok

pedrobaeza commented:
@carlosdauden should I merge or should I wait for your last patches? Can you give your approval?

abhinavvaidya-tsd commented:
is this PR stuck?

carlosdauden commented:
We plan to finish the work by the end of this week.

houzefa-abba reviewed (approved):
Thanks for this work, I tested on runboat / in local env:

* Scan interface works fine with my Inateck BCST-23 without any specific config
* "Picking IN" mode with the "Create lots if not match" setting
* "Picking OUT" mode
* "Show Detailed Operations" setting
* Disable "Fill Fields From Lot" setting to scan the same product several times if multiple quantities are expected
* Play with combos of these settings to scan multiple serial numbers of the same product within 1 picking
* Scan **more** quantities than expected, get "quantities higher than necessary" message, click on red ✔️ to force adding it
* Scan **less** quantities than expected, click "Validate" early on ➔ Partial transfer gets validated & second one is created with remaining quantities

houzefa-abba commented:
Hi I have this issue in inventory mode:
* Give a product a barcode + tracking by serial number
* Go to Barcodes > Inventory
* Scan the product barcode added above

➔ Error:
```
  File "/mnt/data/odoo-addons-dir/stock_barcodes/wizard/stock_barcodes_read.py", line 142, in _compute_qty_available
    done_move_lines = self.move_line_ids.filtered(
AttributeError: 'wiz.stock.barcodes.read.inventory' object has no attribute 'move_line_ids'
```

In 15.0, this displays an "Expected serial number scan" popup, but does not crash.

Also note:
* No error when the product has no tracking by s/n
* No error when scanning a serial number instead

sergio-teruel committed:
[FIX] stock_barcodes: Move qty_available field from read base to read picking

sergio-teruel commented:
Hi @houzefa-abba Fixed in last commit (4978de68945d25d9f6dcf07beb269b11c666817e)

Thanks

alexis-via commented:
Thanks for your great work on the migration of this great module.

I have a usability remark on inventory: you added a new boolean option "display_read_quant" on the wizard wiz.stock.barcodes.read.inventory. When this option is not checked, the user doesn't see the result of the scans, which is error-prone.
So I suggest 3 solutions:
1) modify the definition of this field to set default=True
2) move this option to stock.barcodes.option.group, so that we can enable this option for all inventories
3) add an option on stock.barcodes.option.group that will act as a default value for the field "display_read_quant" of the wizard (combination of solution 1 and 2)
I don't have a strong opinion myself about the best option. As of today, my opinion is : it should not be an option, because I think it's too risky to scan without visibility on the result !
If you want, I'm ready to make a PR to your branch to implement the preferred solution.

alexis-via commented on stock_barcodes/models/stock_quant.py:
```diff
@@ -0,0 +1,31 @@
+            for fname in quant._fields:
+                if fname in self._get_fields_to_edit():
+                    wiz_barcode[fname] = quant[fname]
+            wiz_barcode.product_qty = quant.inventory_quantity
```
When reading this code, I see that, on line 30, in the loop on quant._fields that are in _get_fields_to_edit(), it will write:
```
wiz_barcode.product_qty
```
then, on line 31, it will write:
```
wiz_barcode.product_qty = quant.inventory_quantity
```
Is it normal ? Shouldn't we remove "product_qty" from _get_fields_to_edit ?

sergio-teruel committed:
[IMP] stock_barcodes: Add display read items on inventory mode in option group and minor fix

sergio-teruel commented:
Hi @alexis-via thanks for your comments

I have added display read items in the option group to apply as a default value.
Also fix the view and your suggestion in stock quant

alexis-via commented:
@sergio-teruel I made some tests with your last commit: it works very well, thanks a lot !
Thanks for fixing the view with the display of the quants!

pedrobaeza commented:
/ocabot merge nobump
